test/runners.jl

```diff
@@ -101,7 +103,8 @@ end
     @testset "Compilation - C++ string ABI" begin
         mktempdir() do dir
             # Host is x86_64-linux-musl-cxx11 and target is x86_64-linux-musl-cxx03
-            ur = preferred_runner()(dir; platform=Platform(arch(HostPlatform()), "linux"; libc="musl", cxxstring_abi="cxx03"), preferred_gcc_version=v"5")
+            platform = HostPlatform(parse(Platform, join(["x86_64", split(Base.BinaryPlatforms.host_triplet(), "-")[2:end]...], "-")))
+            ur = preferred_runner()(dir; platform=Platform(arch(platform), "linux"; libc="musl", cxxstring_abi="cxx03"), preferred_gcc_version=v"5")
```

Comment:
It doesn't make any sense to build in a convoluted way a platform whose architecture is always `x86_64` and then at the line below pass only its architecture, forgetting all the rest.  Just use `x86_64` for the architecture of the runner.
```suggestion
            ur = preferred_runner()(dir; platform=Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx03"), preferred_gcc_version=v"5")
```
You just need to undo https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/178/files#diff-be899a28c772a23287ab65459ca9fd65569793ca855fcb68fa63fdf0c78592d5R83